ui/src/app/views/workspace-edit/component.tsx

```diff
@@ -0,0 +1,584 @@
+    height: '16px',
+    marginLeft: '0.2rem',
+    width: '16px'
+  },
+  select: {
+    display: 'inline-block',
+    verticalAlign: 'middle',
+    position: 'relative',
+    overflow: 'visible',
+    width: '11.3rem',
+    marginRight: '20px'
+  },
+  textArea: {
+    width: '100%',
+    marginTop: '0.5rem',
+    height: '10em',
+    resize: 'none',
+    padding: '0.1rem 0.2rem',
+    background: '#fff',
+    border: '1px solid #ccc',
+    color: '#000',
+    borderRadius: '.125rem'
+  },
+  shortDescription: {
+    color: '#262262',
+    fontSize: '16px',
+    fontWeight: 600,
+    lineHeight: '24px'
+  },
+  longDescription : {
+    position: 'relative',
+    display: 'inline-block',
+    minHeight: '1rem',
+    cursor: 'pointer',
+    lineHeight: '1rem',
+    width: '95%'
+  },
+  categoryRow: {
+    display: 'flex',
+    flexDirection: 'row',
+    padding: '0.6rem 0',
+    width: '50%'
+  }
+});
+
+
+export const WorkspaceEditSection = (props) => {
+  return <div key={props.header}>
+    <div style={{display: 'flex', flexDirection: 'row', marginTop: (props.largeHeader ? 48 : 24)}}>
+      <div style={{...styles.header,
+        fontSize: (props.largeHeader ? 20 : 16)}}>
+        {props.header}
+      </div>
+      {props.required && <div style={styles.requiredText}>
+        (Required)
+      </div>
+      }
+      {props.tooltip && <TooltipTrigger content={props.tooltip}>
+        <InfoIcon style={{...styles.infoIcon,  marginTop: '0.2rem'}}/>
+      </TooltipTrigger>
+      }
+    </div>
+    {props.subHeader && <div style={{...styles.header, color: '#4A4A4A', fontSize: 14}}>
+      {props.subHeader}
+    </div>
+    }
+    <div style={styles.text}>
+      {props.text}
+    </div>
+    <div>
+      {props.children}
+    </div>
+  </div>;
+};
+
+export const WorkspaceCategory = (props) => {
+  return <div style={...fp.merge(styles.categoryRow, props.style)}>
+    <input style={{height: '.66667rem', marginRight: '.31667rem'}} type='checkbox'
+           checked={!!props.value}
+           onChange={props.onChange ? (e => props.onChange(e.target.checked)) : undefined}/>
+    <div style={{display: 'flex', flexDirection: 'column', marginTop: '-0.2rem'}}>
+      <label style={styles.shortDescription}>
+        {props.item.shortDescription}
+      </label>
+      <div>
+        <label style={styles.longDescription}>
+          {props.item.longDescription}
+        </label>
+        {props.children}
+      </div>
+    </div>
+  </div>;
+};
+
+export enum WorkspaceEditMode { Create = 1, Edit = 2, Clone = 3 }
+
+
+export interface WorkspaceEditProps {
+  routeConfigData: any;
+  workspace: Workspace;
+}
+
+export interface WorkspaceEditState {
+  cdrVersionItems: Array<CdrVersion>;
+  disableButton: boolean;
+  workspace: Workspace;
+  workspaceCreationConflictError: boolean;
+  workspaceCreationError: boolean;
+  cloneUserRole: boolean;
+  loading: boolean;
+}
+
+export const WorkspaceEdit = withRouteConfigData()(withCurrentWorkspace()(
```

Comment:
nit: If you have multiple HOCs, it's sometimes nicer to compose them with `fp.flow`, which limits the depth of nesting.